File: Practice/pattern19.cpp
// # 5 -> input
// #     1
// #    121
// #   12321
// #  1234321
// # 123454321

#include<iostream>
using namespace std;

int main(void){
int n;
cin>>n;
int space=n-1;
int num=1;


for(int i=0;i<n;i++)
{
    for(int j=0;j<space;j++)
    {
        cout<<" ";
    }
    int count=1;
    for(int k=0;k<num;k++)
    {
        cout<<count;
        if (k<num/2){
            count++;
        }
        else
        {
            count--;
        }
        
    }
    num+=2;
    cout<<endl;
    space--;
}

}

File: Practice/pattern_printing/pattern19.py
# 5 -> input
#     1
#    121
#   12321
#  1234321
# 123454321

num=int(input())

i=1
space=num-i
number=1

while i<=num:
    j=1
    counting=1
    while j<=space:
        print(" ",end=" ")
        j+=1
   
    j=1
    while j<=number:
        print(counting,end="")
        
        if j<=(number//2):
            counting+=1
        else:
            counting-=1
        j+=1
    space-=1
    number+=2
    i+=1
    print()
        


